bl.js

```diff
@@ -40,7 +40,7 @@ BufferList.prototype._offset = function _offset (offset) {
   var tot = 0, i = 0, _t
   for (; i < this._bufs.length; i++) {
     _t = tot + this._bufs[i].length
-    if (offset < _t)
+    if (offset < _t || i == this._bufs.length - 1)
```

Comment:
Because _offset() returned undefined when an index >= length was requested. Most methods of Buffer deal with an index out of range gracefully. I think _offest() should not be concerned with range checking. If we return undefined, even the most simple assumption, that _offset() returns an array of two elements, is broken.